src/caretogether-pwa/src/Model/QueueModel.ts

```diff
@@ -52,11 +60,64 @@ const missingPrimaryContactsQuery = selector<MissingPrimaryContact[]>({
                 adult.item1!.id === family.family?.primaryFamilyContactPersonId
             )
         )
-        .map(
-          (family) =>
-            ({ type: 'MissingPrimaryContact', family }) as MissingPrimaryContact
-        ) || [];
-    return missingPrimaryContacts;
+        .map((family) => ({ type: 'MissingPrimaryContact', family })) || []
+    );
+  },
+});
+
+const childNotReturnedQuery = selector<ChildNotReturned[]>({
+  key: 'childNotReturnedQuery',
+  get: ({ get }) => {
+    const visibleFamilies = get(visibleFamiliesQuery);
+
+    const allArrangements: Arrangement[] =
+      visibleFamilies?.flatMap((family) =>
+        family.partneringFamilyInfo
+          ? [
+              ...(family.partneringFamilyInfo.openReferral?.arrangements || []),
+              ...(family.partneringFamilyInfo.closedReferrals?.flatMap(
+                (referral) => referral.arrangements || []
+              ) || []),
+            ]
+          : []
+      ) || [];
+
+    return allArrangements
+      .filter((arrangement) => arrangement.phase === ArrangementPhase.Ended)
+      .filter((arrangement) => {
+        const mostRecentLocation =
+          arrangement.childLocationHistory &&
+          arrangement.childLocationHistory.length > 0
+            ? arrangement.childLocationHistory[
+                arrangement.childLocationHistory.length - 1
+              ]
+            : null;
+
+        return mostRecentLocation
+          ? mostRecentLocation.plan !== ChildLocationPlan.WithParent
+          : false;
```

Comment:
This part can be simplified like this:

```suggestion
        const mostRecentLocation =
          arrangement?.childLocationHistory?.[
            arrangement.childLocationHistory.length - 1
          ];

        return mostRecentLocation?.plan !== ChildLocationPlan.WithParent;
```